OSDOCS-11169: making CSI optional for microshift (#80823)
openshift/openshift-docs: modules/microshift-disabling-uninstalling-lvms-csi-snapshot.adoc

// Module included in the following assemblies:
//
// * microshift_storage/microshift-storage-plugin-overview.adoc

:_mod-docs-content-type: CONCEPT
[id="microshift-disabling-uninstalling-lvms-csi-snapshot_{context}"]
= Disabling and uninstalling LVMS CSI provider and CSI snapshot deployments

> You can configure {microshift-short} to disable the built-in logical volume manager storage (LVMS) Container Storage Interface (CSI) provider or the CSI snapshot capabilities to reduce the use of runtime resources such as RAM, CPU, and storage. Also, You can uninstall the installed CSI implementations using `OC` commands.

Reviewer: ```suggestion
You can configure {microshift-short} to disable the built-in logical volume manager storage (LVMS) Container Storage Interface (CSI) provider or the CSI snapshot capabilities to reduce the use of runtime resources such as RAM, CPU, and storage. Also, You can uninstall the installed CSI implementations using `oc` commands.
```

Author: done

> You can configure {microshift-short} to disable the built-in logical volume manager storage (LVMS) Container Storage Interface (CSI) provider or the CSI snapshot capabilities to reduce the use of runtime resources such as RAM, CPU, and storage. Also, you can uninstall the installed CSI implementations using `oc` commands.

Reviewer 2: ```suggestion
You can reduce the use of runtime resources such as RAM, CPU, and storage by removing or disabling the following storage components:

* You can configure {microshift-short} to disable the built-in logical volume manager storage (LVMS) Container Storage Interface (CSI) provider. 
* You can disable the CSI snapshot capabilities. 
* You can uninstall the installed CSI implementations using `oc` commands.
```
This is better. Puts the goal first and practical options second.

[IMPORTANT]
====

> Automated uninstallation is not supported as this can cause orphaning of the provisioned volumes. Without the LVMS CSI driver, the cluster goes not have knowledge of the underlying storage interface and cannot perform provisioning and deprovisioning or mounting and unmounting operations.

Author: done

// Module included in the following assemblies:
//
// * microshift_storage/microshift-storage-plugin-overview.adoc

:_mod-docs-content-type: CONCEPT
[id="microshift-disabling-uninstalling-lvms-csi-snapshot_{context}"]
= Disabling and uninstalling LVMS CSI provider and CSI snapshot deployments

You can reduce the use of runtime resources such as RAM, CPU, and storage by removing or disabling the following storage components:

* You can configure {microshift-short} to disable the built-in logical volume manager storage (LVMS) Container Storage Interface (CSI) provider.

> * You can disable the CSI snapshot capabilities.

Reviewer 3: ```suggestion
* You can disable the CSI snapshot capabilities prior to first boot.
```

> * You can uninstall the installed CSI implementations using `oc` commands.

Reviewer 3: The CSI snapshot bullets should be matched by the LVMS bullets:
You disable LVMS...
You can uninstall LVMS...

Also, it's probably good to call out that you can only preemptively disable either component before start microshift for the first time. After that you must update the config AND uninstall the components.

Author: done
